core/headerchain.go

```diff
@@ -413,6 +413,36 @@ func (hc *HeaderChain) SetCurrentHeader(head *types.Header) {
 // each header is deleted.
 type DeleteCallback func(common.Hash)
 
+func (hc *HeaderChain) PurgeAbove(n uint64, delFn DeleteCallback) {
+
+	glog.V(logger.Warn).Infof("Purging block data above #%d", n)
+
+	// Set up logging for block purge progress.
+	ticker := time.NewTicker(time.Second * 5)
+	defer ticker.Stop()
+	go func() {
+		for range ticker.C {
+			glog.V(logger.Warn).Infof("Removed data through #%d", n)
+		}
+	}()
+
+	lastFoundHeaderN := n
+	for head := hc.GetHeaderByNumber(n); head != nil || lastFoundHeaderN > n - 2048; n++ {
```

Comment:
@tzdybal please double check me here 👓

Reply:
Looks good, but for me, it's more readable in form: `n < lastFoundHeaderN + 2048` (for me it's more suggesting that we're scanning 2048 after last found).